src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,157 @@
+    private static StorageDao storageDao;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        Storage.people.clear();
+    }
+
+    @Test
+    void registerUser_WithLoginNull_NotOk() {
+        user.setLogin(null);
+        user.setPassword("valid123");
+        user.setAge(20);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithInvalidLoginLength_NotOk() {
+        user.setLogin("Steve");
+        user.setPassword("valid123");
+        user.setAge(20);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithPasswordNull_NotOk() {
+        user.setLogin("Steve123");
+        user.setPassword(null);
+        user.setAge(20);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithInvalidPasswordLength_NotOk() {
+        user.setLogin("Steve123");
+        user.setPassword("qwert");
+        user.setAge(20);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithAgeNull_NotOk() {
+        user.setLogin("Steve123");
+        user.setPassword("valid123");
+        user.setAge(null);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithInvalidAge_NotOk() {
+        user.setLogin("Steve123");
+        user.setPassword("valid123");
+        user.setAge(15);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithAllInvalidFields_NotOk() {
+        user.setLogin("Steve");
+        user.setPassword("qwert");
+        user.setAge(15);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithInvalidAgeHigherThanIntegerMaxValue_NotOk() {
+        user.setLogin("Steve123");
+        user.setPassword("valid123");
+        user.setAge(Integer.MAX_VALUE + 10);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithAllValidFields_Ok() {
+        user.setLogin("Steve123");
+        user.setPassword("valid123");
+        user.setAge(20);
+        User returnedUser = registrationService.register(user);
+        assertEquals(user, returnedUser);
```

Comment:
After successful registration, verify that the user was indeed added to `Storage.people` directly instead of relying on the `register` method's return value alone. This ensures that the storage is being correctly modified.